docs/asyncapi-file.md

```diff
@@ -3,4 +3,105 @@ title: "AsyncAPI Specification File"
+
+The specification allows you to define the structure and format of your API including the channels the end user can subscribe to and the format of the messages they will be receiving.
+
+The files describing the message-driven API in accordance with the AsyncAPI Specification are represented as JSON objects and conform to the JSON standards. YAML, being a superset of JSON, can be used as well to represent a A2S (AsyncAPI Specification) file.
+
+The specification file when fed as an input to the generator library via the asyncAPI CLI command, shown in the code snippet below, will generate API documentation or generate message-based API boilerplate code.
+
+```bash
+ag asyncapi.yaml ~/my-template
+```
+1. **ag** is the asyncAPI generator
+2. **asyncapi.yaml** is the specification file
+3. **~/my-template** is the template you are using
+
+> :bulb: **Note:** 
+If you haven't set up the asyncapi CLI tool, please refer to our [installation guide](installation.md)
+
+### Use cases of the API definition file
+- Interactive and clear API documentation.
+- Generation of message-based APIs boilerplate code.
+- Validation of messages the client receives using the JSON schema that defines validation rules for your objects in the spec file. 
+- Application of API management policies to messages before they arrive to the broker.
+
+> :bulb: **Remember:** 
+If you are looking to learn how to create an asyncapi specification file or you need to refresh your knowledge about the syntax and structure of the AsyncAPI Specification file, check out our official [documentation](https://www.asyncapi.com/docs/reference/specification/v2.4.0).
+
+In this documentation you'll learn about the inner working of the generator and what happens once the specification file is fed to the generator and how you can use the content of specification file using the asyncAPI document or asyncAPI string in your template.
+
+### The generation process
+The generator library receives the template and AsyncAPI specification file as inputs. To validate that the json/yaml specification file is valid, the generator passes the stringified version of the original specification document to the [parser](parser.md). The parser validates the asyncapi specification using either the openapi schema, raml schema or the avro schema defined in the spec files. If valid, the parser then manipulates the original JSON or YML specification file into functions and properties and bundles them together into an **asyncapiDocument**. The generator library then passes the originalapiString, the asyncapiDocument, and the extra parameters provided by the user to the renderer engine. The renderer engine renders a template where the originalapiString, the asyncapiDocument and the extra parameters are made accessible to the template developer so they access functions that extract some data from the asyncapi file.
+
```

Comment:
We discussed in our GSoD sync call today how this section is a great introduction to your diagram below. Pratik asked if this diagram was too complicated for a beginner, and I offered some ideas in response:
- Since you have separate sections covering how to use templates vs AsyncAPI documents, one option is to divide the diagram into 2 separate diagrams, one for each method.
- In your overall Generator Docs outline you'll be working on for GSoD, you'll eventually have other PRs open to write other sections such as hooks, parsers, etc. You could also tiny mini-diagrams to each one of those sub-sections to illustrate each piece if you like! For example, you could create a diagram that shows what hooks do within the Generator, another for parsers, and so on. 
- This "Generator Process" section covers most of the info displayed in your diagram, except for the Rendering process. I would add another tiny section and header for `The Rendering Process` right underneath.